src/pages/index.tsx

```diff
@@ -49,11 +53,18 @@ export type FlowDefinition = {
   Comment?: string;
 };
 
+function typedJSONParse<T>(value: string | null): T | undefined {
+  if (value) {
+    return JSON.parse(value);
+  }
+  return undefined;
+}
+
 export default function Home() {
   const router = useRouter();
   const searchParams = useSearchParams();
 
-  const [definition, setDefinition] = useState<string | undefined>();
+  const [definition, setDefinition] = useState<FlowDefinition | undefined>();
```

Comment:
Was this erroring before or just giving an `any` type? The `typedJSONParse` is effectively just an abstracted cast which I'm not entirely sure is what we want to be doing here.